### PR TITLE
hw7

### PR DESCRIPTION
**Pull request by lyubavinaju**

### CODE REVIEW

File: src/main/java/com/epam/tc/hw7/JdiSite.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.epam.tc.hw7.pages.MetalsColorsPage;
+import org.openqa.selenium.WebElement;
+
+@JSite("https://jdi-testing.github.io/jdi-light/")
+public class JdiSite {
+
+    public static HomePage homePage;
+    public static MetalsColorsPage metalsColorsPage;
+
+    @Css("form")
```
Comment (cold-flame) [inline src/main/java/com/epam/tc/hw7/JdiSite.java:19]: In JdiSite class need to declare variable WebPage class

File: src/main/java/com/epam/tc/hw7/pages/MetalsColorsPage.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Url("/metals-colors.html")
+@Title("Metal and Colors")
+public class MetalsColorsPage extends WebPage {
+
+    @Css("form")
+    public MetalsColorsForm form;
+
+
+    @Css(".results li")
+    public JList<UIElement> result;
```
Comment (cold-flame) [inline src/main/java/com/epam/tc/hw7/pages/MetalsColorsPage.java:20]: Can you choose more precise class

File: src/test/java/com/epam/tc/hw7/MetalsAndColorsFormTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        header.navigation.select(HeaderMenuData.METALS_COLORS);
+        metalsColorsPage.checkOpened();
+
+        if (formData.vegetablesSelectedByDefault != null) {
+            metalsColorsPage.form.vegetables.select(formData.vegetablesSelectedByDefault);
+        }
+        metalsColorsPage.form.fill(formData);
+        metalsColorsPage.form.submit();
+
+        metalsColorsPage.form.customRadioOdd.is().selected(formData.customRadioOdd);
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/MetalsAndColorsFormTest.java:59]: Use custom verify with 2 forms e.g. assertThat(Form1).isEqualTo(Form2 )

File: src/main/java/com/epam/tc/hw7/custom/LoginForm.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.epam.tc.hw7.custom;
+
+import com.epam.jdi.light.elements.composite.Form;
+import com.epam.jdi.light.elements.pageobjects.annotations.FindBy;
+import com.epam.jdi.light.ui.html.elements.common.TextField;
+import com.epam.tc.hw7.entities.User;
+
+public class LoginForm extends Form<User> {
+    @FindBy(id = "name")
```
Comment (cold-flame) [inline src/main/java/com/epam/tc/hw7/custom/LoginForm.java:9]: Where is 'fullname' field ?

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/tc/hw7/custom/LoginForm.java (hunk 4 above)
Comment (lyubavinaju) [reply src/main/java/com/epam/tc/hw7/custom/LoginForm.java]: In Header

File: src/main/java/com/epam/tc/hw7/entities/MetalsColorsFormData.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package com.epam.tc.hw7.entities;
+
+public class MetalsColorsFormData {
+    public String vegetablesSelectedByDefault;
+    public String customRadioOdd;
```
Comment (cold-flame) [inline src/main/java/com/epam/tc/hw7/entities/MetalsColorsFormData.java:5]: Why you use access modifiers public for this fields? You also create methods (setters) with public access modifiers. Change public -> private for fields

**Review by cold-flame — CHANGES_REQUESTED** [review]

**Review by cold-flame — APPROVED** [review]

**Pull request merged**